### ISSUE TITLE
Lock prompt in interpreter

### ISSUE THREAD

**Issue by Matelasse** [haskell/haskell-mode#1696]
I dont like how we are able to traverse the prompt in Interactive Mode.
`ƛ>` This part of the interpreter should only be readable and not selectable. The current behavior is such that in insert mode you can go to the left of the prompt and this causes headaches. For reference solution see Python's interactive mode, they handle the prompt perfectly.

**Comment by Matelasse** [comment]
@purcell Sorry to tag you, but do you know any way around this?

**Comment by purcell** [comment]
> do you know any way around this?

Nope. `inf-ruby` behaves the same as this, btw. No idea how `dd` magically has that effect in Python, but I don't have bandwidth to look into it, sorry.
